app/ts/library/ethereum.ts

```diff
@@ -0,0 +1,32 @@
+import { BigNumber, ethers } from 'ethers'
+import { Result } from 'ethers/lib/utils.js'
+import { TransactionReceipt, TransactionResponse, TransferTransactionResponse } from '../types.js'
+import { ERC20ABI } from './ERC20ABI.js'
+
+export function isTransferTransaction(txResponse: TransactionResponse): txResponse is TransferTransactionResponse {
+	return txResponse.data.toLowerCase().startsWith('0xa9059cbb')
+}
+
+export function getTransferTokenValue(transactionReceipt: TransactionReceipt) {
+	const erc20Interface = new ethers.utils.Interface(ERC20ABI)
+	const transferLog = transactionReceipt.logs.find(isTransferLog)
+	if (transferLog === undefined) return undefined
+	const logArgs = erc20Interface.parseLog(transferLog).args
+	return isTransferResult(logArgs) ? logArgs.value : undefined
+}
+
+export function isTransferLog(log: ethers.providers.Log) {
+	const [topic] = log.topics
+	const transferTopic = ethers.utils.id('Transfer(address,address,uint256)')
```

Comment:
This is a *little* bit expensive to calculate (involves a hash).  Consider moving this up to a file-local variable (up 3 lines) so it is calculated once rather than every time this function is called.